tests/unit/cag/test__utils.py

```diff
@@ -53,3 +67,398 @@ def test__validate_table_and_column_names_single_table():
 
     # Assert
     metadata._get_single_table_name.assert_called_once()
+
+
+def test_cast_to_datetime64():
+    """Test the ``cast_to_datetime64`` function.
+
+    Setup:
+        - String value representing a datetime
+        - List value with a ``np.nan`` and string values.
+        - pd.Series with datetime values.
+    Output:
+        - A single np.datetime64
+        - A list of np.datetime64
+        - A series of np.datetime64
+    """
+    # Setup
+    string_value = '2021-02-02'
+    list_value = [None, np.nan, '2021-02-02']
+    series_value = pd.Series(['2021-02-02', None, pd.NaT])
+
+    # Run
+    string_out = cast_to_datetime64(string_value)
+    list_out = cast_to_datetime64(list_value)
+    series_out = cast_to_datetime64(series_value)
+
+    # Assert
+    expected_string_output = np.datetime64('2021-02-02')
+    expected_series_output = pd.Series([
+        np.datetime64('2021-02-02'),
+        np.datetime64('NaT'),
+        np.datetime64('NaT'),
+    ])
+    expected_list_output = np.array(
+        [np.datetime64('NaT'), np.datetime64('NaT'), '2021-02-02'], dtype='datetime64[ns]'
+    )
+    np.testing.assert_array_equal(expected_list_output, list_out)
+    pd.testing.assert_series_equal(expected_series_output, series_out)
+    assert expected_string_output == string_out
+
+
+def test_cast_to_datetime64_datetime_format():
```

Comment:
Use `pytest.mark.parametrize` with `datetime_format`. You can then remove this unit test and combine it with the one above, since most of the code is the same